files/common/usr/bin/unpack-image

```diff
@@ -96,22 +117,35 @@ tar -xzf "$UPGRADE_IMAGE_PATH" ||
 
 report_progress_inc 40 "Verifying format."
 
+# check if this is new style upgrade image
+if [[ -f "complete-unpack" ]]; then
+
+	# verify signatures and hashes before hand-off
+	verify_sig $opt_s
+
+	#
+	# This is newstyle upgrade image that brings with it logic to complete
+	# the unpack process. Hand-off the rest of unpack steps to the script within.
+	#
+	report_progress_inc 50 "Handoff unpack to new script."
+
+	"$(pwd)"/complete-unpack ${f_str:+"-f"} ${s_str:+"-s"} ${x_str:+"-x"} ||
+		die "'complete-unpack' hand-off failed"
+
+	# cleanup the temporary directory and exit
+	popd &>/dev/null || die "'popd' failed"
```

Comment:
The flow you're introducing here is awkward to me.. i.e. how we duplicate some of the "post unpack" logic here, and also below, in the case that there isn't a "complete unpack" script available..

I'd rather us do something like this:
```
if [[ -f "complete-unpack" ]]; do
    ... do stuff ...
else
    ... do other stuff ...
fi

popd &>/dev/null || die "'popd' failed"
report_progress_inc 100 "Unpacking successful."
exit 0
```

This way, it's more clear exactly which steps are common to both code paths, and which steps are not common; and we also avoid duplicating the small amount of logic that's common between them.

Reply:
ok I'll update this.